### PR TITLE
16 cryptography models libraries and queries migration

### PR DESCRIPTION
**Pull request by bdrodes**
Migration of C/C++ crypto models, inventory queries and example alerts.

### CODE REVIEW

File: cpp/ql/lib/experimental/crypto/utils/OpenSSL/CryptoAlgorithm.qll [hunk 1]
```diff
@@ -0,0 +1,1986 @@
...
+ * The `target` and `alias` are converted to lowercase to be of a standard form. 
+ */
+predicate customAliases(string target, string alias){
+    exists(Call c | c.getTarget().getName().toLowerCase() = "obj_name_add" |
+      target = c.getArgument(2).getValue().toLowerCase() and alias =  c.getArgument(0).getValue().toLowerCase())
+  }
+  
+  /**
+ * A hard-coded mapping of known algorithm aliases in OpenSSL. 
+ * This was derived by applying the same kind of logic foun din `customAliases` to the
```
Comment (MathiasVP) [inline cpp/ql/lib/experimental/crypto/utils/OpenSSL/CryptoAlgorithm.qll:72]:
```suggestion
 * This was derived by applying the same kind of logic found in `customAliases` to the 
```

File: cpp/ql/lib/experimental/crypto/utils/OpenSSL/CryptoAlgorithm.qll [hunk 2]
```diff
@@ -0,0 +1,1986 @@
...
+
+
+/**
+ * Enumeration of all known crypto algorithms for openSSL
+ * `name` is all lower case (caller's must ensure they pass in lower case)
+ * `nid` is the numeric id of the algorithm,
+ * `normalized` is the normalized name of the algorithm (e.g., "AES128" for "aes-128-cbc")
+ * `algType` is the type of algorithm (e.g., "SYMMETRIC_ENCRYPTION")
+*/
+predicate knownOpenSSLAlgorithm(string name, int nid, string normalized, string algType)
```
Comment (MathiasVP) [inline cpp/ql/lib/experimental/crypto/utils/OpenSSL/CryptoAlgorithm.qll:169]: 🤯

File: cpp/ql/lib/experimental/crypto/utils/OpenSSL/LibraryFunction.qll [hunk 3]
```diff
@@ -0,0 +1,11295 @@
...
+    openSSLAPIFuncName(f.getName())
+    and
+    isPossibleOpenSSLFunction(f)
+}
+
+/**
+ * OpenSSL functions as defined in the OpenSSL docs
+ * https://www.openssl.org/docs/manmaster/man3/
+ */
+predicate openSSLAPIFuncName(string name){
```
Comment (MathiasVP) [inline cpp/ql/lib/experimental/crypto/utils/OpenSSL/LibraryFunction.qll:23]: 🤯

**Review by MathiasVP — COMMENTED** [review]
A couple of comments on the design (and a couple of ad-hoc way-too-detailed comments when I spotted something).

I'm not sure what kind of review process you have on this repo, so please don't let any of this block your PR. I just wanted to highlight a subset of the things I'd like to see fixed before we get this into the official CodeQL repository 😄.

File: python/ql/src/experimental/cryptography/inventory/new_models/SymmetricEncryptionAlgorithms.ql [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @name Symmetric Encryption Algorithms
+ * @description Finds all potential usage of symmetric encryption algorithms using the supported libraries.
+ * @kind problem
+ * @id cpp/quantum-readiness/cbom/symmetric-encryption-algorithms
```
Comment (ropwareJB) [inline python/ql/src/experimental/cryptography/inventory/new_models/SymmetricEncryptionAlgorithms.ql:5]: Typo in @id metadata: using `cpp` when under python namespace.

**Review by ropwareJB — CHANGES_REQUESTED** [review]

**Comment by bdrodes** [comment]
Merged from @ropwareJB's version of this branch that was also merged upstream. https://github.com/microsoft/codeql/pull/19

**Pull request closed without merging**